### PR TITLE
Adding cs and test types in the intro document

### PR DESCRIPTION
**Pull request by TheTestLynx**
Please review - Some is code and some is theoretical information

### CODE REVIEW

File: intro.html [hunk 1]
```diff
@@ -383,11 +413,99 @@ <h3>Let’s start with an example</h3>
...
 <h2>Types of testing</h2>
 
-<p>TODO: Add paragraphs about acceptance testing, performance testing,
- load testing, regression testing, test driven development, and/or
- behavior-driven development (JBehave, Capybara, & Robot Framework),
- with how they relate to Selenium.
-
+<h3>Acceptance testing</h3>
+<p>This type of test is done in order to determine if a product's
+   feature matches it's requirements.
```
Comment (cgoldberg) [inline intro.html:418]: its not it's

File: intro.html [hunk 2]
```diff
@@ -383,11 +413,99 @@ <h3>Let’s start with an example</h3>
...
 
-<p>TODO: Add paragraphs about acceptance testing, performance testing,
- load testing, regression testing, test driven development, and/or
- behavior-driven development (JBehave, Capybara, & Robot Framework),
- with how they relate to Selenium.
-
+<h3>Acceptance testing</h3>
+<p>This type of test is done in order to determine if a product's
+   feature matches it's requirements. 
+   This generally involve the customer's feedback or specification.</p>
```
Comment (cgoldberg) [inline intro.html:419]: involves

File: intro.html [hunk 3]
```diff
@@ -383,11 +413,99 @@ <h3>Let’s start with an example</h3>
...
+<p>For web applications, the automation of this testing can be done
+   directly with Selenium by simulating a user expected behavior in them.
```
Comment (cgoldberg) [inline intro.html:421]: by simulating user expected behavior.

File: intro.html [hunk 4]
```diff
@@ -383,11 +413,99 @@ <h3>Let’s start with an example</h3>
...
+<p>Note: Acceptance testing is a subtype of <b>functional testing</b>,
+   which some people might also refer to.
+</p>
+
+<h3>Functional testing</h3>
+<p>This type of test is done in order to determine if a product's
+  feature functions properly, without issues.</p>
+<p>This generally include: the tests work without errors 
+  (404, exceptions...), in an usable way (right redirections),
+    in an accessible way and matching it's specifications
```
Comment (cgoldberg) [inline intro.html:433]: its

File: intro.html [hunk 5]
```diff
@@ -383,11 +413,99 @@ <h3>Let’s start with an example</h3>
...
+     (see <b>acceptance testing</b> above).</p>
+<p>For web applications, the automation of this testing can be
+   done directly with Selenium by simulating expected returns.
+   This simulation could be done by record/playback or through
+   the different supported languages as explained in this documentation.</p>
+
+<h3>Performance testing</h3>
```
Comment (cgoldberg) [inline intro.html:440]: why include performance testing info at all?  I'd leave this entire section out.

File: intro.html (hunk 1 above)
Comment (TheTestLynx) [reply intro.html:418]: Ouch, thanks!!

File: intro.html (hunk 5 above)
Comment (TheTestLynx) [reply intro.html:440]: It was part of the TODO. Also, Selenium takes part on it, so why not?
Also, I'm adding these changes as part of a commit for the new documentation, please follow up there, if that's OK :)

**Pull request closed without merging**